app/src/test/java/com/example/marvelcomics/ui/screens/main/MainViewModelTest.kt

```diff
@@ -0,0 +1,147 @@
+package com.example.marvelcomics.ui.screens.main
+
+import com.example.marvelcomics.data.model.Characters
+import com.example.marvelcomics.data.model.Comics
+import com.example.marvelcomics.data.model.Creator
+import com.example.marvelcomics.data.model.Creators
+import com.example.marvelcomics.data.model.Data
+import com.example.marvelcomics.data.model.Events
+import com.example.marvelcomics.data.model.Item
+import com.example.marvelcomics.data.model.Result
+import com.example.marvelcomics.data.model.Series
+import com.example.marvelcomics.data.model.Stories
+import com.example.marvelcomics.data.model.Thumbnail
+import com.example.marvelcomics.data.repository.ComicRepository
+import com.example.marvelcomics.data.wrapper.DataOrException
+import com.example.marvelcomics.util.MainCoroutineRule
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(JUnit4::class)
+class MainViewModelTest {
+
+    @get:Rule
+    var mainCoroutineRule = MainCoroutineRule()
+
+    private lateinit var viewModel: MainViewModel
+
+    private var fakeResult = getResult()
+
+    private var fakeData = getData(results = arrayListOf(fakeResult, fakeResult))
+
+    private var fakeComics = getComics(data = fakeData)
+
+    private val fakeDataOrException = getDataOrException(comics = fakeComics)
+
+    private val fakeRepository: ComicRepository = mockk()
+
+
+    @Before
+    fun setup() {
+
+        fakeResult = fakeResult.copy(title = "Spider-man", description = "description")
```

Comment:
You could keep mocking related code in separate method and just call it in `setup()`, especially with bigger tests 😉